internal/nix/nixprofile/profile.go

```diff
@@ -26,11 +28,52 @@ func ProfileListItems(
 	profileDir string,
 ) (map[string]*NixProfileListItem, error) {
 
-	lines, err := nix.ProfileList(writer, profileDir)
+	version, err := nix.Version()
 	if err != nil {
 		return nil, err
 	}
 
+	// if version is >= 2.17, we can use the json output
+	useJSON := vercheck.SemverCompare(version, "2.17") >= 0
+
+	output, err := nix.ProfileList(writer, profileDir, useJSON)
+	if err != nil {
+		return nil, err
+	}
+
+	if useJSON {
+		type ProfileListElement struct {
+			Active      bool     `json:"active"`
+			AttrPath    string   `json:"attrPath"`
+			OriginalURL string   `json:"originalUrl"`
+			Priority    int      `json:"priority"`
+			StorePaths  []string `json:"storePaths"`
+			URL         string   `json:"url"`
+		}
+		type ProfileListOutput struct {
+			Elements []ProfileListElement `json:"elements"`
+			Version  int                  `json:"version"`
+		}
+
+		var structOutput ProfileListOutput
+		if err := json.Unmarshal([]byte(output), &structOutput); err != nil {
+			return nil, err
+		}
+
+		result := map[string]*NixProfileListItem{}
+		for index, element := range structOutput.Elements {
+			result[element.OriginalURL] = &NixProfileListItem{
+				index:             index,
+				unlockedReference: element.OriginalURL,
+				lockedReference:   element.URL,
```

Comment:
@mikeland73 do we think I got this mapping right?

Reply:
No, it's not 1 to 1 on either locked or unlocked. I'm taking a look